api/v1alpha1/kyma_types.go

```diff
@@ -173,11 +179,23 @@ const (
 )
 
 type ModuleStatus struct {
-	// Name is the current deployed module name
+	// Name is the current manifest name
 	Name string `json:"name"`
+	// Namespace is the current manifest namespace
+	Namespace string `json:"namespace"`
+	// GroupVersionKind is used to track the Kind that was created from the ModuleTemplate. This is dynamic to not bind
+	// ourselves to any kind of Kind in the code and allows us to work generic on deletion / cleanup of
+	// related resources to a Kyma Installation.
+	metav1.GroupVersionKind `json:"gvk"`
 
-	// ModuleName is the unique identifier of the module.
-	ModuleName string `json:"moduleName"`
+	// For defines the name of the Module in the Spec that the status is used for.
+	// It can be any kind of Reference format supported by Module.Name.
+	For string `json:"for"`
```

Comment:
Use a preposition `for` as an attribute is a bit unusual here tbh, can we change to a noun word? why not still use the previously defined `ModuleName` for it?

Reply:
I think For is unfitting as well. 👍 Main Reason I do not want to use ModuleName is because we have `Name` already as the Resource and the status is in `.status.moduleStatus`, so the path would be `.status.moduleStatus.ModuleName` for me this kind of kills the purpose of nested keys. IMO we should try to refactor it to `.status.module.name` and then have name/namespace of the manifest in `.status.module.manifest.name/namespace`. WDYT?